src/utils/cryptography/snip12.cairo

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.10.0 (utils/cryptography/snip12.cairo)
+
+use core::hash::HashStateExTrait;
+use hash::{HashStateTrait, Hash};
+use poseidon::PoseidonTrait;
+use starknet::{ContractAddress, get_tx_info};
+
+// selector!(
+//   "\"StarknetDomain\"(
+//    \"name\":\"shortstring\",
+//    \"version\":\"shortstring\",
+//    \"chainId\":\"shortstring\",
+//    \"revision\":\"shortstring\"
+//   )"
+// );
```

Comment:
I wonder if we should just use the macro directly

Reply:
More expensive step(gas)-wise for a constant thing.